File: apps/api/src/modules/video-tags/entities/video-tag.entity.ts
// src/video-tags/entities/video-tag.entity.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  ManyToOne,
  Unique,
  CreateDateColumn,
  Index,
  Column,
  JoinColumn,
} from 'typeorm';
import { Video } from '../../videos/entities/video.entity';
import { Tag } from '../../tags/entities/tag.entity';

@Entity('video_tags')
@Unique(['video', 'tag'])
@Index('idx_video_tags_video_id', ['video'])
@Index('idx_video_tags_tag_id', ['tag'])
export class VideoTag {
  @PrimaryGeneratedColumn()
  id: number;

  @ManyToOne(() => Video, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'video_id' })
  video: Video;

  @ManyToOne(() => Tag, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'tag_id' })
  tag: Tag;

  @CreateDateColumn({ name: 'created_at' })
  createdAt: Date;

  @Column({ name: 'created_by', type: 'int', nullable: true })
  createdBy?: number | null;
}

File: apps/api_go/internal/domain/video_tag_entity.go
package domain

import "time"

// VideoTag entity - maps to 'video_tags' table (junction table with extra field)
type VideoTag struct {
	ID        uint      `gorm:"primaryKey"`
	VideoID   uint      `gorm:"column:video_id;not null;uniqueIndex:idx_video_tag"`
	Video     *Video    `gorm:"foreignKey:VideoID;constraint:OnDelete:CASCADE"`
	TagID     uint      `gorm:"column:tag_id;not null;uniqueIndex:idx_video_tag"`
	Tag       *Tag      `gorm:"foreignKey:TagID;constraint:OnDelete:CASCADE"`
	CreatedAt time.Time `gorm:"column:created_at;autoCreateTime"`
	CreatedBy *uint     `gorm:"column:created_by"`
}

func (VideoTag) TableName() string {
	return "video_tags"
}
